lib/sign_in.dart

```diff
@@ -0,0 +1,87 @@
+import 'package:badiup/models/admin_model.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+
+import 'package:badiup/constants.dart' as constants;
+import 'package:badiup/models/address_model.dart';
+import 'package:badiup/models/customer_model.dart';
+import 'package:badiup/models/user_model.dart';
+import 'package:badiup/models/user_setting_model.dart';
+
+final FirebaseAuth _auth = FirebaseAuth.instance;
+final GoogleSignIn googleSignIn = GoogleSignIn();
+final db = Firestore.instance;
+User currentSignedInUser = User();
+
+Future<String> signInWithGoogle() async {
+  final GoogleSignInAccount googleSignInAccount = await googleSignIn.signIn();
+  final GoogleSignInAuthentication googleSignInAuthentication =
+      await googleSignInAccount.authentication;
+
+  final AuthCredential credential = GoogleAuthProvider.getCredential(
+    accessToken: googleSignInAuthentication.accessToken,
+    idToken: googleSignInAuthentication.idToken,
+  );
+
+  final FirebaseUser user = await _auth.signInWithCredential(credential);
+
+  assert(!user.isAnonymous);
+  assert(await user.getIdToken() != null);
+
+  final FirebaseUser currentUser = await _auth.currentUser();
+  assert(user.uid == currentUser.uid);
+
+  // check if user already exists
+  final DocumentSnapshot userSnapshot =
+    await db.collection(constants.DBCollections.customers)
```

Comment:
Are we storing all the users in this collection? Then, I think the naming is a bit off. Should we name it "users" collection instead?

Reply:
@dragonfly91 Agree with you.